### ISSUE TITLE
About your dataset

### ISSUE THREAD

**Issue by IVparagigm** [jd730/STRG#2]
Hi, because I'm a green hands and I could not get Kinetics dataset, I only can read your code. So there are some questions:
(1) In videodataset.py, class videodataset return a clip and a target in training. I notice that the length of clip is equal to the length of frame_indices which is 10, but in your paper, you select 32 frames as input. So could you tell me where you select 32 frames?
(2)About strg.py, I test other size of input like 1*3*32*224*224, whose batch size is 1 and depth is 32, but the batch size of output of extractor and reducer is 2. If that means I must process all my input with batch_size = 4? Could I use other size of input?
All problems above are just primary but bothering me for several days. I would be very grateful to you if you could help me.
Thanks.

**Comment by jd730** [comment]
Hi @IVparagigm,
1-1)  Could you tell me which script did you run? on my side, the length of `frame_indices` is 32.
1-2) Also, I am not the author of the paper.

2) I cannot understand the size of your input. Maybe there is a font mismatch. does it 1 * 3 * 32 * 224 * 224?

The batch_size is preserved.

![image](https://user-images.githubusercontent.com/9247850/90763622-f4e07c80-e321-11ea-9c1e-0d8b7c75c189.png)

**Comment by jd730** [comment]
I tested on my mac with new conda env and it worked but when I tested on my Linux server (AWS) with a new conda env, the error still occurred.
 
Specifically, there are below package is installed a new env on Linux

```
Package         Version
--------------- -------------------
blessings       1.7
certifi         2020.6.20
chardet         3.0.4
click           7.1.2
configparser    5.0.0
docker-pycreds  0.4.0
gitdb           4.0.5
GitPython       3.1.7
gpustat         0.6.0
gql             0.2.0
graphql-core    1.1
idna            2.10
numpy           1.19.1
nvidia-ml-py3   7.352.0
opencv-python   4.4.0.42
pathtools       0.1.2
pip             20.2.2
promise         2.3
psutil          5.7.2
python-dateutil 2.8.1
PyYAML          5.3.1
requests        2.24.0
sentry-sdk      0.16.5
setuptools      49.6.0.post20200814
shortuuid       1.0.1
six             1.15.0
smmap           3.0.4
subprocess32    3.5.4
urllib3         1.25.10
wandb           0.9.5
watchdog        0.10.3
wheel           0.34.2
```
and the `opencv_python` is from `opencv_python-4.4.0.42-cp37-cp37m-manylinux2014_x86_64.whl`

**Comment by IVparagigm** [comment]
1)in Videodataset.py    class Videodataset    _make_dataset()
`            frame_indices = list(range(segment[0], segment[1]))
            sample = {
                'video': video_path,
                'segment': segment,
                'frame_indices': frame_indices,
                'video_id': video_ids[i],
                'label': label_id
            }`
And I got the annotation like this:
`    "---QUuC4vJs": {
        "annotations": {
            "label": "testifying",
            "segment": [
                84.0,
                94.0
            ]
        },`
so I consider the length of frame indices is 10.  I'm sorry I can't got dataset so I just read your code to infer the size of data.

2)Yes, I first use github, so there are something wrong with my comment format. This is my test code:
`    rois = torch.rand((1, 32, 10, 4))
    inputs = torch.rand((1, 3, 32, 224, 224))
    strg = STRG(model)
    out = strg(inputs, rois)`
When I use this, I got the feauture with 2 * 512 * 7 * 7, but my expectation was getting a output with size of 32 * 512 * 7 * 7 beacuse the depth of input is 32(I sent 32 frames into network). Could you tell me what's wrong with me.

Tnank you very much if you could help me.

**Comment by jd730** [comment]
@IVparagigm 
1) If I remember correctly, that one is from https://github.com/kenshohara/3D-ResNets-PyTorch/blob/master/datasets/videodataset.py and it is for dense-sampling. it finally samples 32 frames on the next stages.
2) what is `model` in your code? Is it `resnet_strg` correctly?

**Comment by IVparagigm** [comment]
1)Yes, what I metioned is in this file.
`        frame_indices = self.data[index]['frame_indices']
        if self.temporal_transform is not None:
            frame_indices = self.temporal_transform(frame_indices)
        clip = self.__loading(path, frame_indices)`
The `self.temporal_transform` make `frame_indices` into the length of 32, right?

2)`resnet_strg` and `resnet` both I have tried. 
In `resnet_strg`, after `self.reducer` I got the output with size of 16 * 512 * 14 * 14. If output with size of  32 * 512 * 7 * 7 is what I want. Is it achievable?
